week4/assignment1/src/pages/signup/Signup.tsx

```diff
@@ -0,0 +1,143 @@
+import { useNavigate } from 'react-router-dom';
+import { useState } from 'react';
+import { AxiosError } from 'axios';
+import { signUp } from '../../services/authService';
+import PageContainer from '../../layouts/pageContainer/PageContainer';
+import Title from '../../components/title/Title'
+import Input from '../../components/input/Input'
+import Button from '../../components/button/Button'
+import * as styles from './Signup.css';
+
+const Signup = () => {
+  const navigate = useNavigate();
+
+  const [step, setStep] = useState<'id' | 'password' | 'nickname'>('id');
```

Comment:
```suggestion
export const enum SignUpStep {
  ID = "id",
  PASSWORD = "password",
  NICKNAME = "nickname",
}

const [step, setStep] = useState<SignUpStep>(SignUpStep.ID);
```

enum을 사용해서 개선해봐도 좋을 것 같네요 ㅎㅎ

https://github.com/toss/frontend-fundamentals/discussions/6